src/TTextEdit.cpp

```diff
@@ -1104,27 +1104,44 @@ void TTextEdit::expandSelectionToWords()
 {
     int yind = mPA.y();
     int xind = mPA.x();
-    for (; xind >= 0; --xind) {
-        if (mpBuffer->lineBuffer.at(yind).at(xind) == QChar::Space
-            || mpHost->mDoubleClickIgnore.contains(mpBuffer->lineBuffer.at(yind).at(xind))) {
-            break;
+
+    // Check if yind is within the valid range of lineBuffer
+    if (yind >= 0 && yind < static_cast<int>(mpBuffer->lineBuffer.size())) {
+        for (; xind >= 0; --xind) {
+            // Ensure xind is within the valid range for the current line
+            if (xind < static_cast<int>(mpBuffer->lineBuffer.at(yind).size())) {
+                const QChar currentChar = mpBuffer->lineBuffer.at(yind).at(xind);
+                if (currentChar == QChar::Space
+                    || mpHost->mDoubleClickIgnore.contains(currentChar)) {
+                    break;
+                }
+            } else {
+                break; // xind is out of bounds, break the loop
+            }
```

Comment:
Again, let's negate this logic to remove one level of indentation:

```suggestion
            if not (xind < static_cast<int>(mpBuffer->lineBuffer.at(yind).size())) {
                break; // xind is out of bounds, break the loop
            }
            
            const QChar currentChar = mpBuffer->lineBuffer.at(yind).at(xind);
            if (currentChar == QChar::Space
                || mpHost->mDoubleClickIgnore.contains(currentChar)) {
                break;
            }
```

Reply:
This doesn't compile unfortunately.

Could you submit these improvements as a separate PR?